protocol.go

```diff
@@ -51,6 +51,14 @@ func (t tcpDialer) DialSqlConnection(ctx context.Context, c *Connector, p *msdsn
 	var ips []net.IP
 	ip := net.ParseIP(p.Host)
 	if ip == nil {
+		// if the dialer has been updated, the dialer may be proxying to a different network, and so the
+		// dialer should be used to connect so the DNS is resolved within the right network
+		if c != nil && c.Dialer != nil {
```

Comment:
>c != nil [](http://example.com/codeflow?start=5&length=8)

I don't think c is ever nil

Reply:
It looks like there is a similar check in the [getDialer](https://github.com/microsoft/go-mssqldb/blob/main/mssql.go#L207) method, but I'm happy to remove the check, if that check is there because the method is called in different circumstances